### ISSUE TITLE
Fix `Unable to delete PDB resource for workflow` error

### ISSUE THREAD

**Issue by alexec** [argoproj/argo#3706]
## Summary 

Unexpected error.

## Diagnostics

```
evel=error msg="Unable to delete PDB resource for workflow." err="poddisruptionbudgets.policy \"canary-8rpjz\" not found" namespace=argo workflow=canary-8rpjz
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
The first error dupes #3217.

**Comment by alexec** [comment]
Available in [v2.10.0-rc7](https://github.com/argoproj/argo/releases/tag/v2.10.0-rc7).

**Comment by alexec** [comment]
Available for testing in [v2.10.0-rc1](https://github.com/argoproj/argo/releases/tag/v2.11.0-rc1).
